Orleans/src/Microsoft.Orleans.Host.Abstractions/ISiloBuilder.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    public interface ISiloBuilder
```

Comment:
Was planning to put silo builder in Microsoft.Orleans.Silo.Abstractions.

A goal of this design is logical separation of actor model, i.e silo and service framework, i.e host.  Silo should be able to be run in any service framework.  Host is our framework as the silo is not much value without a framework to run it in.